otherlibs/threads/stdlib.ml

```diff
@@ -39,6 +39,11 @@ let invalid_arg s = raise(Invalid_argument s)
 
 exception Exit
 
+let try_finally (work: unit -> 'a) (cleanup: unit -> unit) : 'a =
+  let result = (try work () with e -> cleanup (); raise e) in
+  cleanup ();
+  result
+
```

Comment:
Personally, I find
```ocaml
match work () with
| result -> cleanup (); result
| exception e -> cleanup (); raise e
```
easier to read.

Reply:
I agree. I thought of keeping the changes minimal by just moving the old code from `Misc`, but I don't see why not update the style! I'll update the code, and remove the changes to the compiler as you suggested